concourse/client.py

```diff
@@ -342,6 +383,41 @@ def build_events(self, build_id):
         )
         return BuildEvents(response, self)
 
+    @ensure_annotations
+    def trigger_resource_check(self, pipeline_name: str, resource_name: str):
+        url = self.routes.resource_check(pipeline_name=pipeline_name, resource_name=resource_name)
+        # Resource checks are triggered by a POST with an empty JSON-document as body against
+        # the resource's check-url
+        self._post(url, body='{}')
+
+
+class ConcourseApiV3(ConcourseApiBase):
+    '''
+    Implemets Concourse version 3 specific REST api methods
+
+    Not intended to be used outside of this module. Instantiated by factory method 'from_cfg'
+    '''
+    def __init__(self, concourse_cfg: ConcourseConfig, team_name: str, verify_ssl: str):
```

Comment:
why not let the factory method pass-in the approriate routes object? then, the c'tor would not have to be duplicated

Reply:
it's not just the different routes object, but also different a different way of login login that is done in the c'tor